editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java

```diff
@@ -421,12 +422,21 @@ private void preselectCamelVersionForRuntime(String runtimeCamelVersion) {
 		if (UNKNOWN_CAMEL_VERSION.equals(runtimeCamelVersion)) {
 			camelVersionCombo.setEnabled(true);
 		} else {
+			ArrayList<String> compatibleVersions = new ArrayList<>();
 			for (String camelVersion : camelVersionCombo.getItems()) {
```

Comment:
using Java 8 streams we can write it with 3 lines instead of 6:
			List<String> compatibleVersions = Arrays.stream(camelVersionCombo.getItems())
					.filter(camelVersion -> isCompatible(runtimeCamelVersion, camelVersion))
					.collect(Collectors.toList());
but it is as you prefer :-)